### PR TITLE
[#4091] my resources optimization

### PR DESCRIPTION
**Pull request by pkdash**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. The query optimization implemented in this PR, will be noticeable in terms of page load time for the 'my-resources' page by users who have large number of owned resources (~200).
2. For the purpose of testing page load improvement, you have to create at least 100 resources. Then compare the page load time for the 'my-resources' page between this branch and develop branch. You should see better page load time with this branch compared to the develop branch.

### CODE REVIEW

File: hs_core/models.py
Comment (pkdash) [inline hs_core/models.py:2088]: This reduces 1 query per resource.

File: hs_core/models.py
Comment (pkdash) [inline hs_core/models.py:4250]: This reduces 1 query per resource.

File: hs_core/views/utils.py
Comment (pkdash) [inline hs_core/views/utils.py:672]: This reduces 1 query per resource.

File: hs_composite_resource/tests/test_composite_resource.py
Comment (devincowan) [inline hs_composite_resource/tests/test_composite_resource.py:3892]: yep, login is important! :) thanks for fixing this

File: hs_composite_resource/tests/test_composite_resource.py
Comment (devincowan) [inline hs_composite_resource/tests/test_composite_resource.py:4015]: NICE!

File: theme/templates/includes/my-resources-trows.html
Comment (devincowan) [inline theme/templates/includes/my-resources-trows.html:76]: Nit: `generates` twice

**Review by devincowan — APPROVED** [review]
Looks good, one nitpick comment. Thanks for doing this!
I'm waiting to approve until 2.11 release is upon us

File: theme/templates/includes/my-resources-trows.html
Comment (pkdash) [reply theme/templates/includes/my-resources-trows.html]: fixed it.

**Review by devincowan — COMMENTED** [review]
looks good, thanks Pabitra

**Comment by pkdash** [comment]
@devincowan Can you approve again?

**Comment by devincowan** [comment]
@pkdash yep I was waiting till after Keycloak release to approve/merge anything else
But we are going to delay keycloak, so will approve

**Pull request merged**
